docs/developer-docs/latest/development/custom-fields.md

```diff
@@ -0,0 +1,340 @@
+!!!include(developer-docs/latest/development/snippets/custom-field-requires-plugin.md)!!!
+:::
+
+Custom fields must be registered in Strapi's admin panel to be available in the Content-type Builder and the Content Manager.
+
+The `app.customFields` object exposes a `register()` method on the `StrapiApp` instance. This method is used to register custom fields in the admin panel during the plugin's admin [bootstrap lifecycle](/developer-docs/latest/developer-resources/plugin-api-reference/admin-panel.md#bootstrap).
+
+`app.customFields.register()` registers one or several custom field(s) in the admin panel by passing an object (or an array of objects) with the following parameters:
+
+| Parameter                        | Description                                                              | Type                  |
+| -------------------------------- | ------------------------------------------------------------------------ | --------------------- |
+| `name`                           | Name of the custom field                                             | `String`              |
+| `pluginId`<br/><br/>(_optional_) | Name of the plugin creating the custom field<br/><br/>!!!include(developer-docs/latest/development/snippets/custom-field-plugin-optional.md)!!!                          | `String`              |
+| `type`                           | Existing Strapi data type the custom field will use                  | `String`              |
+| `icon`<br/><br/>(_optional_)     | Icon for the custom field                                            | `React.ComponentType` |
+| `intlLabel`                      | Translation for the name                                             | [`IntlObject`](https://formatjs.io/docs/react-intl/) |
+| `intlDescription`                | Translation for the description                                      | [`IntlObject`](https://formatjs.io/docs/react-intl/) |
+| `components`                     | Components needed to display the custom field in the Content Manager (see [components](#components))            |
+| `options`<br/><br/>(_optional_)  | Options to be used by the Content-type Builder (see [options](#options)) | `Object` |
+
+::: details Example: Registering an example "color" custom field in the admin panel:
+
+```jsx
+// path: ./src/plugins/color-picker/strapi-admin.js
+
+register(app) {
+  app.customFields.register({
+    name: "color",
+    pluginId: "color-picker", // the custom field is created by a color-picker plugin
+    type: "text", // the color will be stored as text
+    intlLabel: {
+      id: "color-picker.color.label",
+      defaultMessage: "Color",
+    },
+    intlDescription: {
+      id: "color-picker.color.description",
+      defaultMessage: "Select any color",
+    } 
+    icon: ColorIcon,
+    components: {
+      Input: async () => import(/* webpackChunkName: "input-component" */ "./Input"),
+      View: async () => import(/* webpackChunkName: "view-component" */ "./View"),
```

Comment:
The View component won't be part of the first iteration of the feature. I think it's fine to leave it in the RFC but it shouldn't be in the docs

Reply:
Noted, thank you. So it would mean `components` can only include one component with the first iteration (i.e. `Input`), right? 
I will adjust the Components section of the documentation then.